k9mail/src/main/java/com/fsck/k9/controller/MessagingController.java

```diff
@@ -2291,6 +2294,46 @@ private void processPendingExpunge(PendingCommand command, Account account)
         }
     }
 
+    private void processPendingExpungeMessage(PendingCommand command, Account account)
```

Comment:
There is some code duplication here,  might be easier to teach processPendingExpunge itself to respect extra arguments? (unless a "no arguments means expunge all" pattern is something we want to avoid, ping @cketti)